docs/notebooks/failure_ego.pct.py

```diff
@@ -107,42 +108,66 @@ def observer(x):
 num_init_points = 15
 initial_data = observer(search_space.sample(num_init_points))
 
+
 # %% [markdown]
-# ## Modelling the data
+# ## Model the data
 #
-# We'll model the data on the objective with a regression model, and the data on which points failed with a classification model. The regression model will be a `GaussianProcessRegression` wrapping a GPflow `GPR`, and the classification model a `VariationalGaussianProcess` wrapping a GPflow `VGP` with Bernoulli likelihood. We'll train both models with L-BFGS-based optimizers.
+# We'll model the data on the objective with a regression model, and the data on which points failed with a classification model. The regression model will be a `GaussianProcessRegression` wrapping a GPflow `GPR`, and the classification model a `VariationalGaussianProcess` wrapping a GPflow `VGP` with Bernoulli likelihood.
 
 # %%
 def create_regression_model(data):
     variance = tf.math.reduce_variance(data.observations)
-    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=0.2 * np.ones(2,))
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=0.2 * np.ones(2, ))
     gpr = gpflow.models.GPR(astuple(data), kernel, noise_variance=1e-5)
     set_trainable(gpr.likelihood, False)
     return gpr
 
 
 def create_classification_model(data):
-    variance = tf.math.reduce_variance(data.observations)
-    kernel = gpflow.kernels.SquaredExponential(variance, lengthscales=0.1 * np.ones(2,))
+    kernel = gpflow.kernels.SquaredExponential(variance=100., lengthscales=0.2 * np.ones(2, ))
     likelihood = gpflow.likelihoods.Bernoulli()
     vgp = gpflow.models.VGP(astuple(data), kernel, likelihood)
+    set_trainable(vgp.kernel.variance, False)
     return vgp
 
 
 regression_model = create_regression_model(initial_data[OBJECTIVE])
 classification_model = create_classification_model(initial_data[FAILURE])
 
+# %% [markdown]
+# ## Create a custom optimize method
+# The new `NatGradTrainedVGP` class has a custom `optimize` method that alternates between Adam steps to optimize the lengthscales and NatGrad steps to optimize the variational parameters:
+
+# %%
+class NatGradTrainedVGP(VariationalGaussianProcess):
+    def __init__(self, model):
+        super().__init__(model)
+        self._learning_rate = 1e-3
+        self._gamma = 0.1
+        self._maxiter = 100
+
+    def optimize(self):
+        set_trainable(self.model.q_mu, False)
+        set_trainable(self.model.q_sqrt, False)
+        variational_params = [(self.model.q_mu, self.model.q_sqrt)]
+        adam_opt_for_vgp = tf.optimizers.Adam(self._learning_rate)
+        natgrad_opt = NaturalGradient(gamma=self._gamma)
+
+        for step in range(self._maxiter):
+            natgrad_opt.minimize(self.model.training_loss, var_list=variational_params)
+            adam_opt_for_vgp.minimize(self.model.training_loss, var_list=self.model.trainable_variables)
```

Comment:
oh i more meant something like the following, which is terser (important for tutorials?)
```suggestion
class NatGradTrainedVGP(VariationalGaussianProcess):
    def optimize(self):
        set_trainable(self.model.q_mu, False)
        set_trainable(self.model.q_sqrt, False)
        variational_params = [(self.model.q_mu, self.model.q_sqrt)]
        adam_opt_for_vgp = tf.optimizers.Adam(1e-3)
        natgrad_opt = NaturalGradient(gamma=0.1)

        for step in range(100):
            natgrad_opt.minimize(self.model.training_loss, var_list=variational_params)
            adam_opt_for_vgp.minimize(self.model.training_loss, var_list=self.model.trainable_variables)
```

Reply:
also why `adam_opt_for_vgp` not `adam_opt`?